### PR TITLE
fix(launchDarkly): ldClient Initialization

### PR DESCRIPTION
**Pull request by Gauravudia**
## PR Description

- Bug reported by Open source contribution [here](https://github.com/rudderlabs/rudder-sdk-js/issues/818)
- [This](https://github.com/rudderlabs/rudder-sdk-js/pull/543) PR has introduced a logic change in v1.1. Here we are using `window.user` while initializing the LaunchDarkly client which is never set.
- In [v1](https://github.com/rudderlabs/rudder-sdk-js/blob/04ec98c984da16856034e4eae8971713de436875/src/integrations/LaunchDarkly/browser.js#L46) we are correctly initializing the LaunchDarkly client with `this.launchDarklyUser`.

## Notion ticket

https://www.notion.so/rudderstacks/LaunchDarkly-Integration-now-fails-with-No-user-specified-bf809b28f1b74c20a77c5f8ffbf26ada

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by mihir-4116 — APPROVED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

File: __tests__/integrations/LaunchDarkly/browser.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  mockAnonymousId,
+  mockUserId,
+  mockClientSideId,
+  mockAnonymousUsersSharedKey,
+  trackCallPayload,
+  mockProperties,
+  testEvent,
+} from './__mocks__/data';
+
+beforeAll(() => {});
```
Comment (MoumitaM) [inline __tests__/integrations/LaunchDarkly/browser.test.js:14]: I believe this is not needed.

File: __tests__/integrations/LaunchDarkly/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import createUser from '../../../src/integrations/LaunchDarkly/utils';
+import { mockAnonymousUsersSharedKey, mockTraits, mockUserId } from './__mocks__/data';
+
+beforeAll(() => {});
```
Comment (MoumitaM) [inline __tests__/integrations/LaunchDarkly/utils.test.js:4]: Same here. Remove this line.

File: __tests__/integrations/LaunchDarkly/browser.test.js (hunk 1 above)
Comment (MoumitaM) [reply __tests__/integrations/LaunchDarkly/browser.test.js:14]: Line 14

File: __tests__/integrations/LaunchDarkly/utils.test.js (hunk 2 above)
Comment (MoumitaM) [reply __tests__/integrations/LaunchDarkly/utils.test.js:4]: line 4

File: __tests__/integrations/LaunchDarkly/browser.test.js (hunk 1 above)
Comment (Gauravudia) [reply __tests__/integrations/LaunchDarkly/browser.test.js]: updated

File: __tests__/integrations/LaunchDarkly/utils.test.js (hunk 2 above)
Comment (Gauravudia) [reply __tests__/integrations/LaunchDarkly/utils.test.js]: updated

**Review by MoumitaM — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**